backend/app/api/endpoints/dataset_items.py

```diff
@@ -0,0 +1,137 @@
+# Copyright (C) 2025 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+from datetime import datetime
+from typing import Annotated
+from uuid import UUID
+
+from fastapi import APIRouter, Depends, File, HTTPException, Query, UploadFile, status
+from starlette.responses import StreamingResponse
+
+from app.api.dependencies import get_dataset_item_id, get_dataset_item_service
+from app.schemas import DatasetItem, DatasetItemsWithPagination
+from app.schemas.base import Pagination
+from app.services import DatasetItemService, ResourceNotFoundError
+
+router = APIRouter(prefix="/api/dataset/items", tags=["Dataset items"])
+
+DEFAULT_DATASET_ITEMS_NUMBER_RETURNED = 10
+MAX_DATASET_ITEMS_NUMBER_RETURNED = 100
+
+
+@router.post(
+    "",
+    status_code=status.HTTP_201_CREATED,
+    responses={status.HTTP_201_CREATED: {"description": "Dataset item created", "model": DatasetItem}},
+)
+async def create_dataset_item(
+    dataset_item_service: Annotated[DatasetItemService, Depends(get_dataset_item_service)],
+    file: Annotated[UploadFile, File()],
+) -> DatasetItem:
+    """Create a new dataset item"""
```

Comment:
```suggestion
async def add_dataset_item(
    dataset_item_service: Annotated[DatasetItemService, Depends(get_dataset_item_service)],
    file: Annotated[UploadFile, File()],
) -> DatasetItem:
    """Add a new item to the dataset by uploading an image"""
```
Slightly rephrased to highlight that this endpoint is used to upload images